core/src/main/java/feign/AsyncFeign.java

```diff
@@ -255,15 +267,18 @@ protected AsyncFeign(AsyncBuilder<C> asyncBuilder) {
     this.activeContext = new ThreadLocal<>();
 
     this.defaultContextSupplier = asyncBuilder.defaultContextSupplier;
-    this.client = asyncBuilder.client;
 
-    this.logLevel = asyncBuilder.logLevel;
-    this.logger = asyncBuilder.logger;
+    this.client = asyncBuilder.client instanceof AsyncClient.SyncBased
```

Comment:
Why not enrich a `PseudoAsync` client?!

Reply:
So far, I have not found a nicer way to distinguish between "true" async clients (Java11, HC5, OkHttp) and those that wrap client + executor. And that's why I added this `SyncBased` interface. Users don't need to worry about it unless they implement their own clients.

Reply:
But how can you `enrich` the underlying client?

Reply:
Makes sense... From the builder perspective, a user sets `Client` for sync and `AsyncClient` for async. Then feign applies capabilities when the whole thing is being built. It does not know any details of async implementation, meaning there should be no enrichments before that. I'll return to the drawing board. Thanks!

Reply:
Sweet, if  you could just remove the `SyncBased`, I would love to include this on todays release

Reply:
I removed it locally, but I don't think this change is ready for prime time. I've been looking at why I was thinking about double-enrichment in the first place...

In `AsyncFeign`, we keep some functionality in the class, but we also delegate some of it to `Feign.Builder`. Like 

```
public AsyncBuilder<C> invocationHandlerFactory(InvocationHandlerFactory invocationHandlerFactory) {
  builder.invocationHandlerFactory(invocationHandlerFactory);
  return this;
}
```

So when I added this:
```
public AsyncBuilder<C> addCapability(Capability capability) {
  builder.addCapability(capability);
  this.capabilities.add(capability);
  return this;
}
```
It provides capabilities to both Async and delegate (Sync).

When `AsyncFeign` is constructed, we call:
```
  asyncBuilder.builder.client(this::stageExecution);
```
It replaces `client` in the delegate(!) with the keep-this-future function. 

When we call `this.feign = asyncBuilder.builder.build();` a few lines later, we enrich `this::stageExecution` like it's a client because delegate capabilities are invoked in `.build()`. That does not make sense, though, as `this::stageExecution` is not a client but a future-keeper. And that's how we get double-enrichment.

Reply:
shouldn't we let the Capability implementation handle that?

Another alternative, we could change the regular `Builder` to no enrich the client if it is `Async`

Reply:
> Another alternative, we could change the regular Builder to no enrich the client if it is Async

That's what I attempted in the latest commit. Looks similar to `forceDecoding` flag.